src/App.js

```diff
@@ -1,5 +1,81 @@
+const { Console } = require('@woowacourse/mission-utils');
+const { makeRandomNumber, chkDuplicatedNumber } = require('./util');
+
 class App {
-  play() {}
+  #gameAnswer = null;
+
+  static printResult(result) {
+    let output = '';
+    if (result.ball !== 0) {
+      output += `${result.ball}볼`;
+      if (result.strike !== 0) {
+        output += ' ';
+      }
+    }
+    if (result.strike !== 0) {
+      output += `${result.strike}스트라이크`;
+    }
+    if (result.strike === 0 && result.ball === 0) {
+      output = '낫싱';
+    }
+    Console.print(output);
+  }
+
+  static getResult(number, answer) {
+    const numberArr = [...number];
+    const result = numberArr.reduce(
+      ({ strike, ball }, num, nowInd) => {
+        const ind = answer.indexOf(num);
+        if (ind === -1) {
+          return { strike, ball };
+        }
+        if (ind === nowInd) {
+          return { strike: strike + 1, ball };
+        }
+        return { strike, ball: ball + 1 };
+      },
+      { strike: 0, ball: 0 }
+    );
+    return result;
+  }
+
+  decideRestart() {
+    Console.print('3개의 숫자를 모두 맞히셨습니다! 게임 종료');
+    Console.readLine(
+      '게임을 새로 시작하려면 1, 종료하려면 2를 입력하세요.',
+      finishInput => {
+        if (finishInput === '1') {
+          this.#gameAnswer = makeRandomNumber(3, 1, 9);
+          this.inputByConsole();
+        }
+        if (finishInput === '2') {
+          Console.close();
+        }
+      }
+    );
+  }
+
+  inputByConsole() {
+    Console.readLine('숫자를 입력해주세요 : ', input => {
+      if (chkDuplicatedNumber(input, 3))
+        throw new Error(
+          '입력형식이 잘못되었습니다. 서로 다른 숫자 3개를 입력해주세요.🙏'
+        );
+      const result = App.getResult(input, this.#gameAnswer);
+      App.printResult(result);
+      if (result.strike === 3) {
+        this.decideRestart();
+      } else {
+        this.inputByConsole();
+      }
+    });
+  }
```

Comment:
콜백을 외부로 분리해주셔야 해당 메소드의 이름이 제 역할을 할 것 같아요!!
지금의 메소드 명을 유지하신다면, 아래처럼은 어떠신가요?

```suggestion
// 외부로 분리 
const INPUT_NUMBER = '숫자를 입력해주세요 : ';

inputByConsole(callback) {
    Console.readLine(INPUT_NUMBER, callback);
  }

  try(input) {
      if (chkDuplicatedNumber(input, 3))
        throw new Error(
          '입력형식이 잘못되었습니다. 서로 다른 숫자 3개를 입력해주세요.🙏'
        );
      const result = App.getResult(input, this.#gameAnswer);
      App.printResult(result);
      if (result.strike === 3) {
        this.decideRestart();
      } else {
        this.inputByConsole();
      }
    }
```

```js
//호출시 
this.inputByConsole(this.try);
```

Reply:
앗 이부분 분리해야지 하다가 못했네요 ㅎㅎ^_^

Reply:
ㅎㅎ 종종 저도 그럴 떄가 있더라구요..😂